server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -96,4 +86,21 @@ class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticC
       )
     }
   }.map(response => PostcodeAddressFileAddresses(response.as[PostcodeAddressFileAddress], response.maxScore))
+
```

Comment:
Writing unreadable code is worse then some repeated code (especially if it's only one repetition)
Exposing these arguments wont do any good, the query will change and will be very different for NAG and PAF so it would be better to put a query as a parameter. But there are also index name and a callback with different classes as well.